File: core/src/main/java/forpleuvoir/dhwu1a/core/event/friend/FriendEventType.java
package forpleuvoir.dhwu1a.core.event.friend;

/**
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.event.friend
 * <p>#class_name FriendEventType
 * <p>#create_time 2021/7/3 20:59
 */
public enum FriendEventType {
    FriendInputStatusChangedEvent(FriendInputStatusChangedEvent.class),
    FriendNickChangedEvent(FriendNickChangedEvent.class),
    FriendRecallEvent(FriendRecallEvent.class);


    private final Class<? extends FriendEvent> clazz;

    FriendEventType(Class<? extends FriendEvent> clazz) {
        this.clazz = clazz;
    }

    public Class<? extends FriendEvent> getClazz() {
        return clazz;
    }

    public static boolean hasKey(String key) {
        var isHas = false;
        for (FriendEventType value : FriendEventType.values()) {
            isHas = value.name().equals(key);
            if (isHas) break;
        }
        return isHas;
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/event/friend/FriendEventType.kt
package forpleuvoir.dhwu1a.core.event.friend

/**
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.event.friend
 *
 * #class_name FriendEventType
 *
 * #create_time 2021/7/3 20:59
 */
enum class FriendEventType(val clazz: Class<out FriendEvent?>) {
    FriendInputStatusChangedEvent(forpleuvoir.dhwu1a.core.event.friend.FriendInputStatusChangedEvent::class.java),
    FriendNickChangedEvent(forpleuvoir.dhwu1a.core.event.friend.FriendNickChangedEvent::class.java),
    FriendRecallEvent(forpleuvoir.dhwu1a.core.event.friend.FriendRecallEvent::class.java);

    companion object {
        fun hasKey(key: String): Boolean {
            var isHas = false
            for (value in values()) {
                isHas = value.name == key
                if (isHas) break
            }
            return isHas
        }
    }
}